### PR TITLE
fix: rich editor image insertion

### PR DESCRIPTION
**Pull request by bobmitch**
Insert image in rich editor currently broken on flat, this is an attempt to get it working.
Will need to attempt a better DRY approach that both this and the Image field can both use.

### CODE REVIEW

File: core/fields/Field_Rich.php [hunk 1]
```diff
@@ -583,14 +591,22 @@ function makeLinkHTML(href, content) {
 								//alert('choose image');
 								// launch image selector
 								var media_selector = document.createElement('div');
+								media_selector.id = "editor_media_selector";
```
Comment (workaj) [inline core/fields/Field_Rich.php:594]: is this repeatable friendly?

**Review by workaj — COMMENTED** [review]
didnt checkout the branch and test yet, but one thought

File: core/fields/Field_Rich.php [hunk 2]
```diff
@@ -637,30 +653,148 @@ function makeLinkHTML(href, content) {
 								// search handler
 								let searchtrigger = document.getElementById('trigger_media_selector_search').addEventListener('click',function(e){
 									let searchtext = document.getElementById('media_selector_modal_search').value;
-									fetch_images(searchtext, null); // string, no tags
+									rich_fetch_images(searchtext, null); // string, no tags
 								});
 
+								window.cur_media_page = 1;
+								window.cur_media_searchtext = null;
+								window.images_per_page = ;
```
Comment (workaj) [inline core/fields/Field_Rich.php:661]: this obviously doeesnt work at all

**Review by workaj — CHANGES_REQUESTED** [review]
bug

File: core/fields/Field_Rich.php (hunk 2 above)
Comment (bobmitch) [reply core/fields/Field_Rich.php:661]: lol

**Review by workaj — APPROVED** [review]
works

**Pull request merged**